### PR TITLE
refactor: initial table adjustments

### PR DESCRIPTION
**Pull request by ItsANameToo**
## Summary

https://app.clickup.com/t/861mruby8

Note that this also affects the delegate voting table and all mobile tables will now have semibold labels

TODO:
- [x] handle address truncation below `xl` screens
- [x] determine wallet ranks
- [x] add dark/ark to balance table header
- [x] remove token suffix from balances
- [x] add additional rows to mobile view
- [x] add info icon to percentage column
- [x] dark mode
- [x] check if wallet votes look alright as they use the same table
- [x] cleanup by moving into separate components

## Checklist

-   [ ] I checked my UI changes against the design and there are no notable differences
-   [ ] I checked my UI changes for any responsiveness issues
-   [ ] I checked my UI changes in light AND dark mode
-   [ ] I checked my (code) changes for obvious issues, debug statements and commented code
-   [ ] I opened a corresponding card on Clickup for any remaining TODOs in my code
-   [ ] I added a short description on how to test this PR _(if necessary)_
-   [ ] Documentation _(if necessary)_
-   [ ] Tests _(if necessary)_
-   [ ] Ready to be merged

### CODE REVIEW

File: resources/views/components/tables/desktop/wallets.blade.php [hunk 1]
```diff
@@ -1,30 +1,68 @@
...
+            <x-tables.headers.desktop.number name="general.wallet.rank" />
             <x-tables.headers.desktop.address name="general.wallet.address" />
-            <x-tables.headers.desktop.icon name="general.wallet.info" class="text-center" />
-            <x-tables.headers.desktop.number name="general.wallet.balance" />
-            <x-tables.headers.desktop.number name="general.wallet.supply" />
+            <x-tables.headers.desktop.text name="general.wallet.name" />
+            <x-tables.headers.desktop.icon name="general.wallet.type" class="text-center" />
+            @if(!($hideVoting ?? false))
+                <x-tables.headers.desktop.icon name="general.wallet.voting" class="text-center" responsive breakpoint="lg" />
+            @endif
```
Comment (alexbarnsley) [inline resources/views/components/tables/desktop/wallets.blade.php:10]:
```suggestion
            @unless($hideVoting ?? false)
                <x-tables.headers.desktop.icon name="general.wallet.voting" class="text-center" responsive breakpoint="lg" />
            @endunless
```

File: resources/views/components/tables/desktop/wallets.blade.php [hunk 2]
```diff
@@ -1,30 +1,68 @@
...
+
                 <x-ark-tables.cell class="text-center">
                     <x-tables.rows.desktop.wallet-type :model="$wallet" />
                 </x-ark-tables.cell>
+
+                @if(!($hideVoting ?? false))
+                    <x-ark-tables.cell class="text-center" responsive breakpoint="lg" >
+                        <x-tables.rows.desktop.encapsulated.voting :model="$wallet" />
+                    </x-ark-tables.cell>
+                @endif
```
Comment (alexbarnsley) [inline resources/views/components/tables/desktop/wallets.blade.php:42]:
```suggestion
                @unless($hideVoting ?? false)
                    <x-ark-tables.cell class="text-center" responsive breakpoint="lg" >
                        <x-tables.rows.desktop.encapsulated.voting :model="$wallet" />
                    </x-ark-tables.cell>
                @endunless
```

File: resources/views/components/tables/desktop/wallets.blade.php [hunk 3]
```diff
@@ -1,30 +1,68 @@
-<x-ark-tables.table sticky class="hidden w-full md:block">
-    <thead>
-        <tr>
+<x-tables.encapsulated-table sticky class="hidden w-full md:block">
```
Comment (alexbarnsley) [inline resources/views/components/tables/desktop/wallets.blade.php:1]: Can we do this or is it some magic component? :thinking: 

```suggestion
@props([
    'wallets',
    'hideVoting' => false,
    'useVoteWeight' => false, // change the `isset($userVoteWeight)` checks
])

<x-tables.encapsulated-table sticky class="hidden w-full md:block">
```

File: resources/views/components/tables/encapsulated-table.blade.php [hunk 4]
```diff
@@ -0,0 +1,6 @@
+
+<div {{ $attributes->class(["rounded-t-xl border border-theme-secondary-300 dark:border-theme-secondary-800 table-container px-6 table-encapsulated encapsulated-table-header-gradient"]) }}>
```
Comment (alexbarnsley) [inline resources/views/components/tables/encapsulated-table.blade.php:2]:
```suggestion
<div {{ $attributes->class(["rounded-t-xl border border-theme-secondary-300 dark:border-theme-secondary-800 table-container px-6 table-encapsulated encapsulated-table-header-gradient"]) }}>
```

File: resources/views/components/tables/rows/desktop/address.blade.php [hunk 5]
```diff
@@ -1,5 +1,5 @@
-<x-general.identity :model="$model" :without-truncate="$withoutTruncate ?? false">
-    @if ($model->username())
+<x-general.identity :model="$model" :without-truncate="$withoutTruncate ?? false" :without-username="$withoutUsername ?? false">
+    @if ($model->username() && !($withoutUsername ?? false))
```
Comment (alexbarnsley) [inline resources/views/components/tables/rows/desktop/address.blade.php:2]: As before, can we add props?

```suggestion
@props([
    'model',
    'withoutUsername' => false,
    'withoutTruncate' => false,
])

<x-general.identity :model="$model" :without-truncate="$withoutTruncate" :without-username="$withoutUsername">
    @if ($model->username() && !$withoutUsername)
```

File: resources/views/components/tables/rows/desktop/encapsulated/address.blade.php [hunk 6]
```diff
@@ -0,0 +1,15 @@
+<span class="flex justify-between w-full">
```
Comment (alexbarnsley) [inline resources/views/components/tables/rows/desktop/encapsulated/address.blade.php:1]:
```suggestion
@props([
    'model',
    'withoutUsername' => false,
    'withoutTruncate' => false,
])

<span class="flex justify-between w-full">
```

Comment (alexbarnsley) [reply resources/views/components/tables/rows/desktop/encapsulated/address.blade.php:1]: I'll stop posting the props for the other files :trollface:

**Review by alexbarnsley — CHANGES_REQUESTED** [review]
Code review:
i've mentioned `@props` in a few files but not all of them, i suspect it's intentional as the table stuff might be some magic components. disregard those comments if needs be

Implementation review:
Percentage icon should be an `i`

Design:
image

Impl
image

----------------

Spacing between heading and subheading is too large (this can be skipped as it's not a major issue)

Design
image

Impl
image

--------------

Mobile should have dashed border separator

Design
image

Impl
image

**Comment by alexbarnsley** [comment]
FYI [this](https://github.com/ArdentHQ/arkscan/pull/202/commits/c2403decc96bd2c67db68e616e35bad3647493e8) fixes the gap between header & subtitle

File: resources/views/components/tables/desktop/wallets.blade.php (hunk 3 above)
Comment (ItsANameToo) [reply resources/views/components/tables/desktop/wallets.blade.php:1]: props will work, just kept it in line with what we had before but will adjust it to make it prettier and easier to work with

**Pull request merged**